internal/io/file/file_sink.go

```diff
@@ -234,6 +235,22 @@
 					}
 				}
 			}
+			if len(m.c.Fields) > 0 {
```

Comment:
how about just let headers = fields? When writing, make sure all fields are have value, even it is nil